base/set.jl

```diff
@@ -101,8 +101,15 @@ function in!(x, s::Set)
 end
 
 push!(s::Set, x) = (s.dict[x] = nothing; s)
-pop!(s::Set, x) = (pop!(s.dict, x); x)
```

Comment:
Could you just return the result[1] here?

Reply:
Popping a dict only returns the value, not the key, value pair

Reply:
I didn't realize that. It looked like we did it a few lines later

Reply:
Ah, I see. It can't be fixed now, but this seems awkward (partly for the same reason that motivated this PR--the popped key is not necessarily exactly the same as the argument)
```
julia> pop!(Dict(1=>2), 1)
2

julia> pop!(Dict(1=>2))
1 => 2
```